src/plugins/resolver/filename.c

```diff
@@ -193,7 +193,37 @@ static int elektraResolveSystem (char variant, resolverHandle * p, Key * warning
 		elektraResolveFinishByFilename (p);
 		return 1;
 	}
+	if (p->path[0] == '~')
+	{
+		ssize_t bufSize = sysconf (_SC_GETPW_R_SIZE_MAX);
+		if (bufSize == -1) bufSize = 16384; // man 3 getpwuid
+
+		char * buf = elektraMalloc (bufSize);
+		if (!buf) return -1;
+		struct passwd pwd;
+		struct passwd * result;
+		int s;
+
+		s = getpwuid_r (getuid (), &pwd, buf, bufSize, &result);
```

Comment:
I am not sure if this is useful because it only checks for the current user, not for every possible user. You should at least avoid this code duplication.